tests/unit/services/apollo-test.js

```diff
@@ -4,26 +4,23 @@ import ApolloService from 'ember-apollo-client/services/apollo';
 
 const { computed } = Ember;
 
-let options;
-
 moduleFor('service:apollo', 'Unit | Service | apollo', {
-  beforeEach() {
-    options = {
-      apiURL: 'https://test.example/graphql',
-    };
-  },
+  needs: ['config:environment'],
 });
 
 test('it exists', function(assert) {
+  let options = {
+    apiURL: 'https://test.example/graphql',
+  };
   let service = this.subject({ options });
   assert.ok(service);
 });
 
 test('it uses clientOptions', function(assert) {
   let customDataIdFromObject = o => o.name;
   let OverriddenService = ApolloService.extend({
-    // Need this here because apollo requires a uri, but our initializer doesn't
-    // run in unit tests.
+    // Need this here because apollo requires a uri, but our config environment
+    // isn't set up in unit tests.
     options: {
       apiURL: 'https://this-should-be-set-from-environment.example',
     },
```

Comment:
It looks like `OverriddenService` doesn't have an owner because you're `create()`-ing one manually.

Instead, you could do something like:

```js
let customDataIdFromObject = o => o.name;
this.register('service:custom-apollo', ApolloService.extend({
  clientOptions: // ...
}));
let service = this.container.lookup('service:custom-apollo');
// ...
```

Reply:
you mean just like any other service test? 🤦‍♂️ 

Thanks as always!